src/judoscale/django/middleware.py

```diff
@@ -28,5 +29,14 @@ def __call__(self, request):
 
             metrics_store.add(metric)
             logger.debug("queue_time={}ms".format(round(queue_time_ms, 2)))
+            self._get_reporter()
 
         return self.get_response(request)
+
+    def _get_reporter(self):
+        try:
+            if not reporter._thread.is_alive():
```

Comment:
I'm confused how this works. If `reporter.start()` has not yet been called, I think `reporter._thread` will be `None`. I would expect calling `is_alive()` on `None` to raise an exception.

Do you see what I'm missing?

Reply:
That's a valid point. The thing is that the thread starts with the application preload, and then once gunicorn starts the worker this thread is stopped. I added some logging info on the thread itself to verify:

> 9:53:47 AM web.1   |  INFO - [Judoscale] Installing middleware
> 9:53:47 AM web.1   |  INFO - [Judoscale] Starting reporter for process 16188
> 9:53:47 AM web.1   |  DEBUG - [Judoscale] Posting 0 metrics from 16188 to Judoscale adapter API https://judoscale-python.requestcatcher.com/v1/metrics
> 9:53:47 AM web.1   |  INFO - [Judoscale] **THREAD is <Thread(Thread-1, started daemon 140240101857024)> inside Reporter**
> 9:53:47 AM web.1   |  [2022-07-28 06:53:47 +0000] [16188] [INFO] Starting gunicorn 20.1.0
> 9:53:47 AM web.1   |  [2022-07-28 06:53:47 +0000] [16188] [INFO] Listening at: http://0.0.0.0:5100 (16188)
> 9:53:47 AM web.1   |  [2022-07-28 06:53:47 +0000] [16188] [INFO] Using worker: sync
> 9:53:47 AM web.1   |  [2022-07-28 06:53:47 +0000] [16238] [INFO] Booting worker with pid: 16238
> 9:53:48 AM proxy.1 |  Listening on port 5000
> 9:53:48 AM proxy.1 |  Proxying requests to port 5100
> 9:53:53 AM web.1   |  DEBUG - [Judoscale] Posting 0 metrics from 16188 to Judoscale adapter API https://judoscale-python.requestcatcher.com/v1/metrics
> 9:54:05 AM web.1   |  DEBUG - [Judoscale] queue_time=7.6ms
> 9:54:05 AM web.1   |  INFO - [Judoscale] **THREAD is <Thread(Thread-1, stopped daemon 140240101857024)> inside Middleware**
> 9:54:05 AM web.1   |  INFO - [Judoscale] Starting reporter for process 16238
> 9:54:05 AM web.1   |  DEBUG - [Judoscale] Posting 1 metrics from 16238 to Judoscale adapter API https://judoscale-python.requestcatcher.com/v1/metrics
> 9:54:05 AM web.1   |  INFO - [Judoscale] **THREAD is <Thread(Thread-2, started daemon 140240101857024)> inside Reporter**
> 9:54:05 AM web.1   |  Hello, world

Reply:
But we should check if thread is alive also before posting of the metrics so we don't get these logs of _posting 0 metrics..._